gnomad_hail/utils/relatedness.py

```diff
@@ -445,3 +446,51 @@ def discard_multi_parents_children(trios: List[hl.Trio]):
         trio for fam_index, parent_child_pairs in enumerate(parent_child_pairs_by_fam)
         for trio in get_trios(str(fam_index), parent_child_pairs, related_pairs)
     ])
+
+
+def create_fake_pedigree(
+    n: int, sample_list: List[str], real_pedigree: hl.Pedigree = None
+) -> hl.Pedigree:
+    """
+    Generates a pedigree made of trios created by sampling 3 random samples in the sample list.
+    If `real_pedigree` is given, then children from the real pedigrees won't be used as probands.
+    This functions insures that:
+    - All probands are unique
+    - All individuals in a trio are different
+
+    :param n: Number of fake trios desired in the pedigree
+    :param sample_list: List of samples
+    :param real_pedigree: Optional pedigree to exclude children from
+    :return: Fake pedigree
+    """
+
+    probands = set()
+    if real_pedigree is not None:
+        probands = {trio.s for trio in real_pedigree.trios}.intersection(
+            set(sample_list)
+        )
+        if len(probands) == len(sample_list):
```

Comment:
Should be something like len(set(probands) == len(set(sample_list) to account for duplicates in sample_list. Something to take into consideration is that the pedigrees we pull from seqr report trios with the s, mat_id, and pat_id columns filled out, but the parents are also output on their own lines with blank mat_id and pat_ids columns, so I think I got stuck in the while loop because I tried feeding in a seqr pedigree file as input. Does the real_pedigree here have to be complete trios only?

Reply:
hmmm, I suppose not; I'll modify this so it can accept incomplete pedigrees. I suppose the only constraint really is that we do not reconstruct an existing true trio